pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SortOperator.java

```diff
@@ -175,6 +175,14 @@ private void consumeInputBlocks() {
               _rows.addAll(container.subList(0, _numRowsToKeep - _rows.size()));
             }
           }
+          // '_fetch > 0' means this operator can early terminate, so we can stop consuming input blocks
+          if (_rows.size() >= _numRowsToKeep && _fetch > 0) {
```

Comment:
We don't allow unlimited rows. Checking `if (_rows.size() == _numRowsToKeep)` is good enough

Reply:
updated.